### PR TITLE
Add resource config and introduce factories

### PR DESCRIPTION
**Pull request by wachterjohannes**

### CODE REVIEW

File: src/Factory/CreditMemoFactory.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            $comment,
+            $issuedAt,
+            $from,
+            $to
+        );
+    }
+
+    public function createNew()
+    {
+        throw new \RuntimeException();
```
Comment (wachterjohannes) [inline src/Factory/CreditMemoFactory.php]: the factory has to implement the `FactoryInterface::createNew` although its not possible to create a `CreditMemo` without constructor parameter.

If the factory does not the ResouceController throws an exception.

File: src/Factory/RefundFactory.php [hunk 2]
```diff
@@ -14,4 +14,12 @@ public function createWithData(string $orderNumber, int $unitId, int $amount, Re
     {
         return new Refund($orderNumber, $amount, $unitId, $type);
     }
+
+    /**
+     * @return object
+     */
+    public function createNew()
+    {
+        throw new \RuntimeException();
```
Comment (wachterjohannes) [inline src/Factory/RefundFactory.php]: the factory has to implement the `FactoryInterface::createNew` although its not possible to create a `Refund` without constructor parameter.

If the factory does not the ResouceController throws an exception.

File: .travis.yml [hunk 3]
```diff
@@ -45,6 +45,7 @@ jobs:
             before_script:
                 - (cd tests/Application && bin/console doctrine:database:create -vvv)
                 - (cd tests/Application && bin/console doctrine:migrations:migrate -n -vvv -q)
+                - (cd tests/Application && bin/console d:s:u --force -q)
```
Comment (wachterjohannes) [inline .travis.yml]: should i add a migration file for this SQLs?

**Comment by wachterjohannes** [comment]
@Zales0123 are you able to trigger a build on master? i think the issue i have in the tests should also happen there

**Comment by GSadee** [comment]
I'm closing this PR as `Refund`, `RefundPayment` and `CreditMemo` implement now the `ResourceInterface`

**Pull request closed without merging**